### PR TITLE
Feat: Infinite scrolling

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following-
- Added a custom hook that increases page number and observes if the last element is visible
- Added loader when the last element is visible and to simulate the action of more posts being loaded.

### CODE REVIEW

File: src/hooks/useInfiniteScroll.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef, useState } from "react";
+
+const useInfiniteScroll = (posts) => {
+	const [pageNumber, setPageNumber] = useState(1);
+	const lastElementReference = useRef(null);
+	const postsLength = posts.length;
+	const hasMorePosts = pageNumber <= Math.ceil(postsLength / pageNumber);
```
Comment (parthpandyappp) [inline src/hooks/useInfiniteScroll.jsx:7]: This looks interesting. Does this have any special name? I've never used this arrow in reversed manner.

**Comment by parthpandyappp** [comment]
The animated loader in the infinite scrolling looks so good :fire:

The code looks good to me, Go ahead with the merge :+1:

**Review by parthpandyappp — APPROVED** [review]

**Review by MonikaJhingan — APPROVED** [review]
The code looks clean and logic looks good. Use of hooks has been done nicely.

**Comment by bharati-21** [comment]
> The animated loader in the infinite scrolling looks so good 🔥
> 
> The code looks good to me, Go ahead with the merge 👍

Thank you for your review!

**Comment by bharati-21** [comment]
> The code looks clean and logic looks good. Use of hooks has been done nicely.

Thank you for your review!

**Pull request merged**